### PR TITLE
Sub-components are added in onInitialize

### PR DESCRIPTION
**Pull request by solomax**

### CODE REVIEW

File: bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/button/BootstrapButton.java [hunk 1]
```diff
@@ -49,11 +49,18 @@ public BootstrapButton(final String componentId, final Buttons.Type type) {
     public BootstrapButton(final String componentId, final IModel<String> model, final Buttons.Type type) {
         super(componentId, model);
 
-        add(buttonBehavior = new ButtonBehavior(type, Buttons.Size.Medium));
+        buttonBehavior = new ButtonBehavior(type, Buttons.Size.Medium);
+
+        icon = newIcon("icon");
```
Comment (martin-g) [inline bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/button/BootstrapButton.java:54]: This needs to be improved. Calling overrideable methods in the constructor is not good.
Do you want to improve it or should I do it ?

Comment (solomax) [reply bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/button/BootstrapButton.java:54]: Would appreciate if you will do it :)

Comment (solomax) [reply bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/button/BootstrapButton.java:54]: @martin-g there seems to be lots of places for improvement
Maybe you can you can show me what do you have in mind and I'll prepare PR?

Comment (martin-g) [reply bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/button/BootstrapButton.java:54]: Sorry! I was busy with other things!
Improved it!

**Pull request merged**